connectivity/FEATURE_BLE/source/generic/PrivateAddressController.cpp

```diff
@@ -0,0 +1,670 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PrivateAddressController.h"
+
+namespace ble {
+
+PrivateAddressController::PrivateAddressController(
+    PalPrivateAddressController &address_generator,
+    PalEventQueue &event_queue,
+    ble::resolvable_address_timeout_t rotation_timeout
+) :
+    _pal(address_generator) ,
+    _event_queue(event_queue),
+    _rotation_timeout(rotation_timeout)
+{
+    _pal.initialize();
+    _pal.set_event_handler(this);
+
+    auto** next = &_free_resolution_entries;
+    for (auto &entry : _resolution_list) {
+        *next = &entry;
```

Comment:
what is the point of this loop? if the entries already point in a chain why go over them in a loop?